src/test/integrationTest/authRoute.test.js

```diff
@@ -0,0 +1,94 @@
+/* eslint-disable camelcase */
+import chai, { expect, should } from 'chai';
+import chaiHTTP from 'chai-http';
+import server from '../..';
+
+should();
+chai.use(chaiHTTP);
+const route = '/api/v1/auth/signup';
+const generateNumber = () => `${Math.round(Math.random() * 1000)}${Date.now()}`;
+const validUser = {
+  email: `john${generateNumber()}@doe.com`,
+  password: 'this1sN0TmyPassword'
+};
+
+const shortPassword = {
+  email: `john${generateNumber()}@doe.com`,
+  password: 'short'
+};
+
+const nonAlphaNumPassword = {
+  email: `john${generateNumber()}@doe.com`,
+  password: 'isThisInvalid!? :-)'
+};
+
+const invalidEmail = {
+  email: 'invalidEmail',
+  password: 'noNeedForPasswordAnyMore'
+};
+
+const nullEmail = {
+  password: 'noNeedForPasswordAnyMore'
+};
+
+const nullPassword = {
+  email: `john${generateNumber()}@doe.com`
+};
+
+const runTest = (done, object = {}) => {
+  chai
+    .request(server)
+    .post(route)
+    .send(object)
+    .then((res) => {
+      res.should.have.status(400);
+      res.body.should.be.an('object');
+      res.body.should.have.property('error');
+      const { status } = res.body;
+      expect(status).to.eql(res.status);
+      done();
+    })
+    .catch(done);
+};
+
+describe('USER VALIDATION TEST', () => {
+  before('Register a valid user', (done) => {
+    chai
+      .request(server)
+      .post(route)
+      .send(validUser)
+      .then((res) => {
+        res.should.have.status(201);
+        done();
+      })
+      .catch(done);
+  });
+
+  it('should throw error 400 when user signs up with existing email', (done) => {
+    runTest(done, validUser);
+  });
+
+  it('should throw error 400 when user signs up with a short password', (done) => {
+    runTest(done, shortPassword);
+  });
+
+  it('should throw error 400 when user signs up with a non-alphanumeric password', (done) => {
+    runTest(done, nonAlphaNumPassword);
+  });
+
+  it('should throw error 400 when user signs up with an invalid email', (done) => {
+    runTest(done, invalidEmail);
+  });
+
+  it('should throw error 400 when user signs up with a null email', (done) => {
+    runTest(done, nullEmail);
+  });
+
+  it('should throw error 400 when user signs up with a null password', (done) => {
+    runTest(done, nullPassword);
+  });
+
+  it('should throw error 400 when user signs up with a null email and null password', (done) => {
+    runTest(done);
+  });
+});
```

Comment:
you would need to re-write your tests because I don't understand the tests you wrote.

Reply:
Since all my tests are the same code, instead of repeating it, I encapsulated it in a function.
Line 37
```
const runTest = (done, object = {}) => {
  chai
    .request(server)
    .post(route)
    .send(object)
    .then(res => {
      expect(res).to.have.status(400);
      const { body } = res;
      expect(body).to.be.an('object');
      expect(body).to.have.property('error');
      const { status } = body;
      expect(status).to.eql(res.status);
      done();
    })
    .catch(done);
};
```